src/object/interrupt.c

```diff
@@ -217,9 +225,10 @@ void handleInterrupt(irq_t irq)
             printf("Undelivered IRQ: %d\n", (int)IRQT_TO_IRQ(irq));
 #endif
         }
-#ifndef CONFIG_ARCH_RISCV
+#if !defined(CONFIG_ARCH_RISCV) && !defined(CONFIG_ARM_GIC_V3_SUPPORT)
         maskInterrupt(true, irq);
 #endif
```

Comment:
We don't do this much in the kernel usually, but would it work to us an actual `if` instead of condition compilation here? Then we can follow both paths in the verification and it'll work either way. IIRC, the compile should optimise out any `if (0)` cases.

Reply:
e.g.
```c
if (!config_set(CONFIG_ARCH_RISCV) && !config_set(CONFIG_ARM_GIC_V3_SUPPORT)) {
  maskInterrupt(true, irq);
}
```

(for that the function has to exist with this signature in those configs, which I haven't checked)

Reply:
That would work, but what do the proofs assume? I'd have a look at the IRQ abstract model, because all we have now is `maskInterrupt` AFAIK, but reality is more complex than that. Does the proof know about individual IRQ states currently?

Reply:
It does, yes. It at least keeps track of what is masked on the IRQ controller side and what not. I would need to check to see if there is more.

Reply:
Then you have no choice other than extending the model to cope with how HW actual work for verified platforms.

Reply:
Yes we'll have to do that when we get to gicv3, but we would still have more trouble with conditional compilation (`#ifdef`) vs control flow (normal `if`). If gicv3 was forever outside verification it wouldn't matter, then `#ifdef` is not bad, because we never see it. But if we want one proof that can do both GIC versions, we need to be able to see both control flow paths.

It's possible that this rabbit hole is too deep, we can look at all this more properly when we actually get to the verification of GICv3. It'll actually be easier in the short term if it's an `#ifdef`, just wanted to get some feedback on if there would be problems from the kernel side with using `if` and `config_set`. I've searched through the code a bit more, and we do seem to use it a lot more than I initially thought, so it seems generally fine.

Reply:
If changing the model you would have an unconditional call for both here. One would do nothing for GICv3, and perhaps another function would do nothing for GICv2. I suspect you want to replace maskInterrupt and ackInterrupt with more fine grained versions that make it clearer what the intention is, that would hopefully also get rid of the RISCV ifdefs.